services/121-service/src/payments/fsp-integration/intersolve-visa/intersolve-visa.api.service.ts

```diff
@@ -515,35 +515,36 @@ export class IntersolveVisaApiService {
     return;
   }
 
-  public async createPhysicalCard(
-    createPhysicalCardDto: CreatePhysicalCardDto,
-  ): Promise<void> {
+  public async createPhysicalCard(input: CreatePhysicalCardDto): Promise<void> {
     // Create the request body to send
-    const createPhysicalCardRequestDto = new CreatePhysicalCardRequestDto();
-    createPhysicalCardRequestDto.firstName = ''; // in 121 first name and last name are always combined into 1 "name" field, but Intersolve requires first name, so just give an empty string
-    createPhysicalCardRequestDto.lastName = createPhysicalCardDto.name;
-    createPhysicalCardRequestDto.mobileNumber = formatPhoneNumber(
-      createPhysicalCardDto.phoneNumber,
-    ); // TODO: Removed the check-for-null statement, since formatPhoneNumber already throws an error if the phone number is null
-    createPhysicalCardRequestDto.cardAddress = {
-      address1:
-        `${createPhysicalCardDto.addressStreet} ${createPhysicalCardDto.addressHouseNumber} ${createPhysicalCardDto.addressHouseNumberAddition}`.trim(),
-      city: createPhysicalCardDto.addressCity,
-      country: 'NLD',
-      postalCode: createPhysicalCardDto.addressPostalCode,
-    };
-    createPhysicalCardRequestDto.pinAddress = {
-      address1:
-        `${createPhysicalCardDto.addressStreet} ${createPhysicalCardDto.addressHouseNumber} ${createPhysicalCardDto.addressHouseNumberAddition}`.trim(),
-      city: createPhysicalCardDto.addressCity,
-      country: 'NLD',
-      postalCode: createPhysicalCardDto.addressPostalCode,
+    // TODO: As per new good practices: this request structure is not a DTO anymore, but created inline. Do the same for the request objects in all other methods in this class. Or should they be interfaces?
+    const request = {
+      brand: 'VISA_CARD',
+      firstName: '',
+      lastName: input.name,
+      // TODO: We need to add a "+" for Intersolve's API to work. Do we have a generic helper function for this?
+      mobileNumber: '+' + input.phoneNumber, // must match \"([+]){1}([1-9]){1}([0-9]){5,14}\"
+      cardAddress: {
+        address1:
+          `${input.addressStreet} ${input.addressHouseNumber} ${input.addressHouseNumberAddition}`.trim(),
+        city: input.addressCity,
+        country: 'NLD',
```

Comment:
Are we sure this is correct? I find it weird that for createCustomer we use 'NL' and for this one 'NLD'.

Reply:
Yes, that is weird. And yes, this is correct. I changed it to NL and got an error back.